app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,56 @@
+class PostsController < ApplicationController
+	
+  before_action :authenticate_user!
+  before_action :set_post, only: [:show, :edit, :update, :destroy]
+
+  def show
+    @comments = @post.comments
+  end
+
+  def index
+    @posts = Post.all
+  end
+
+  def new
+    @post = Post.new
+  end 
+
+  def create
+    @post = current_user.posts.build(post_params)
+    if @post.save
+      redirect_to posts_path, notice: 'Post was successfully created.'
+    else
+      render :new
+    end
+  end
+
+  def edit
+    if @post.update(edit_params)
```

Comment:
Guys, I think the `edit` action isn't where we should be placing this line. Please see `update` action below, it is doing the exact same thing (@post.update). The `edit` action's job isn't to `update` the post, it's merely to show the edit form. 

```ruby
def edit; end
```